core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -65,12 +72,20 @@ public TermSuggestionBuilder(String name) {
      * tokens in the suggest text.
      * </ol>
      */
-    public TermSuggestionBuilder suggestMode(String suggestMode) {
+    public TermSuggestionBuilder suggestMode(@Nullable SuggestMode suggestMode) {
```

Comment:
I don't think we should allow null's to be passed into Builders as this can easily lead to NPEs. If this settings are not required then the user should not set them rather than setting them to null.